File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Fury.java

package com.watabou.pixeldungeon.actors.buffs;

import com.nyrds.pixeldungeon.ml.R;
import com.nyrds.platform.util.StringsManager;
import com.watabou.pixeldungeon.actors.Char;
import com.watabou.pixeldungeon.actors.CharUtils;
import com.watabou.pixeldungeon.sprites.CharSprite;
import com.watabou.pixeldungeon.ui.BuffIndicator;
import com.watabou.pixeldungeon.utils.GLog;

import org.jetbrains.annotations.NotNull;

public class Fury extends Buff {
	
	public static float LEVEL	= 0.4f;
	
	@Override
	public boolean act() {
		if (target.hp() > target.ht() * LEVEL) {
			detach();
		}
		
		spend( TICK * 10 );
		
		return true;
	}

	@Override
	public boolean attachTo(@NotNull Char target) {
		if (CharUtils.isVisible(target)) {
			GLog.w(StringsManager.getVar(R.string.Brute_Enraged), target.getName() );
			target.showStatus( CharSprite.NEGATIVE, StringsManager.getVar(R.string.Brute_StaEnraged));
		}
		return super.attachTo(target);
	}

	@Override
	public int attackProc(Char attacker, Char defender, int damage) {
		return damage * 2;
	}

	@Override
	public int icon() {
		return BuffIndicator.FURY;
	}
}

File: scripts/buffs/Fury.lua
---
--- Fury: doubled attack damage while below 40% hp (was actors/buffs/Fury.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

local CharSprite = luajava.bindClass("com.watabou.pixeldungeon.sprites.CharSprite")

return buff.init{
    desc  = function ()
        return {
            icon          = 18, -- BuffIndicator.FURY
            name          = "FuryBuff_Name",
            info          = "FuryBuff_Info",
        }
    end,

    act = function(self, buff)
        local target = buff.target
        if target:hp() > target:ht() * 0.4 then
            buff:detach()
        end
        buff:spend(10) -- Actor.TICK * 10
    end,

    attackProc = function(self, buff, defender, damage)
        return damage * 2
    end,

    attachVisual = function(self, buff)
        local target = buff.target
        RPD.GLog:w(RPD.JavaUtils:format(RPD.textById("Brute_Enraged"), { target:getName() }), {})
        target:showStatus(CharSprite.NEGATIVE, RPD.textById("Brute_StaEnraged"))
    end
}
